File: festival-mvc/src/main/java/cn/bdqfork/mvc/context/MethodInvocation.java
package cn.bdqfork.mvc.context;

import java.io.Serializable;
import java.lang.reflect.Method;

/**
 * @author bdq
 * @since 2020/1/26
 */
public class MethodInvocation implements Serializable {
    private String methodName;
    private Class<?>[] argumentClasses;
    private Object[] arguments;

    public MethodInvocation() {
    }

    public MethodInvocation(Method method, Object[] arguments) {
        this.methodName = method.getName();
        this.argumentClasses = method.getParameterTypes();
        this.arguments = arguments;
    }

    public String getMethodName() {
        return methodName;
    }

    public void setMethodName(String methodName) {
        this.methodName = methodName;
    }

    public Class<?>[] getArgumentClasses() {
        return argumentClasses;
    }

    public void setArgumentClasses(Class<?>[] argumentClasses) {
        this.argumentClasses = argumentClasses;
    }

    public Object[] getArguments() {
        return arguments;
    }

    public void setArguments(Object[] arguments) {
        this.arguments = arguments;
    }
}

File: festival-kotlin-web/src/main/kotlin/cn/bdqfork/kotlin/web/service/MethodInvocation.kt
package cn.bdqfork.kotlin.web.service

import java.io.Serializable
import java.lang.reflect.Method

/**
 * @author bdq
 * @since 2020/1/26
 */
class MethodInvocation(method: Method, var arguments: Array<Any>) : Serializable {
    var methodName: String = method.name
    var argumentClasses: Array<Class<*>> = method.parameterTypes

}